### ISSUE TITLE
Drop BOM

### ISSUE THREAD

**Issue by mcandre** [PowerShell/PowerShell#8636]
PowerShell silently inserts a Unicode BOM at the beginning of output streams such as with `echo`, causing downstream processes to fail. For example, if you try to use `echo` to author a pip.ini file, pip will be unable to load the file due to the BOM.

**Comment by vexx32** [comment]
I thought PowerShell Core uses largely the BOM-less UTF8 variant for a majority of its standard encodings. :thinking:

Can you please verify this is happening with PowerShell Core, and not Windows PowerShell?

**Comment by chuanjiao10** [comment]
Delete your outdated ideas from your brain please. 
Ini is a text file, any text file must with bom, otherwise it is not conducive to storing unicode characters.

Writing files with both **echo** and **“>，<，>>，|"**, etc.
 It is a stupid practice from unix*. Because writing to a file is not necessary to transfer from the screen output.

The correct way is to use:
Add-Content
Clear-Content
Get-Content
Set-Content
```powershell
$a = 'you string'
Add-Content -LiteralPath /tmp/a.ini -Value $a -Encoding utf8BOM # with bom
Add-Content -LiteralPath /tmp/a.ini -Value $a -Encoding  utf8NoBOM # without  bom
```
Add-Content  be same with ">>"
Set-Content  be same with ">"

**Comment by IISResetMe** [comment]
@mcandre can you provide a short sample that reproduces the behavior you describe?

**Comment by Jaykul** [comment]
@vexx32 the help for Set-Content says the default encoding is ascii?

@chuanjiao10 text files don't need a BOM. Byte order is meaningless in UTF-8, and [isn't even recommended(http://www.unicode.org/faq/utf_bom.html#bom5). In fact, if the OP was correct that _streams_ were getting a BOM, this would probably be an egregious error (i.e if `echo` (aka `Write-Output`) was adding a BOM, that would be bad.

However, I'm pretty sure that's not the case.  Additionally PowerShell has been using utf-8 without a preamble (aka BOM) as the default `$OutputEncoding` since [the 6.0 release candidate](https://github.com/PowerShell/PowerShell/releases/tag/v6.0.0-rc) [PR #5369](https://github.com/PowerShell/PowerShell/pull/5369), so my guess is that @mcandre is dealing with Windows PowerShell 5.x or older.

**Comment by chuanjiao10** [comment]
@Jaykul 
Text files do not require a BOM. ----10 years ago the wrong idea. OR This person has not used unicode, this person will not use unicode in the text file in the future.

By the same token, Linux 10 years ago did not need to rely on bom. But now linux, with centos7 set to Chinese as an example, man page is a Chinese string. If you don't use bom to lock the encoding type, it will generate garbled characters.

in conclusion:
The current linux distribution man page is multilingual and requires a bom. Any text file that does not use bom unicode must be garbled.

Don't think you don't need bom, don't suggest others to remove bom.

**Comment by chuanjiao10** [comment]
In my eyes, this is a war between language types in pc and server.
The battlefield is win, linux, mac. Since various languages occupy windows, the main battlefield is linux now.
Army 1 is an English language.
Army 2 is an Asian language. The people who make up the army are Hindi, Chinese, Japanese, and Korean.
The general in the English army, (engonly Trump) shouted: "go go go ！Discard BOM header in text file, english first !"

**Comment by IISResetMe** [comment]
There's a lot of talk of war in here, but I still don't see any code to reproduce the issue originally described - @mcandre can you enlighten us?

**Comment by Amaroq-Clearwater** [comment]
@chuanjiao10 Listen. UTF-8 is the modern standard for text-encoding. And being a mixed-case encoding format, it does not need a BOM. A lot of programs work just fine using heuristics to correctly identify unicode characters in UTF-8 strings, and UTF-8 is for the most part backwards-compatible with ANSI.

On the other hand, a lot of Windows programs mistakenly insert a BOM into UTF-8 text, Notepad included, and other Windows programs such as Command Prompt will actually fail to correctly parse UTF-8 strings because they'd try to parse a BOM as if it were normal text, thus breaking a command.

**Comment by chuanjiao10** [comment]
@Amaroq-Clearwater 
Do you know that unilateralism vs. multilateralism?
Utf-8 is not the only one, but you and Trump will say "only utf8 is right",
will say "Compatible with english is first" ---- Do you know that English is not the top three languages in the world?
will say "no need bom, guess strings encoding is right"

**Comment by Amaroq-Clearwater** [comment]
Do you have a single fact to back that up?

With all due respect, you're being arrogant, stubborn, and needlessly hostile. I'm only going to politely ask you one more time to stop.

**Comment by Amaroq-Clearwater** [comment]
Back on subject, I agree with the original poster that PowerShell (and Notepad, also) needs to stop inserting a BOM when encoding UTF-8, but PowerShell also needs to be given the ability to ignore a BOM when it encounters one in a UTF-8 string.

**Comment by SteveL-MSFT** [comment]
Please respect our [Code of Conduct](https://github.com/PowerShell/PowerShell/blob/master/CODE_OF_CONDUCT.md) and stay on topic.

As for this issue, PSCore6 defaults to UTF-8 BOM-less encoding for output, so unless a repro can be produced, this seems to be about Windows PowerShell which is not serviced by this repo.
